All_home_works/hw1/task4.py

```diff
@@ -0,0 +1,32 @@
+"""
+Given a list of integers numbers "nums".
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    queue = []
+    max_sum = nums[0]
+
+    for num in nums:
+        if len(queue) == k:
+            queue.pop(0)
```

Comment:
What is the complexity of this line?

Reply:
O(n)

Reply:
But what complexity if queue from collections will be used (It will be linked list. So it means O(1))?

Reply:
Sorry for grammar)

Reply:
> Sorry for grammar)

Try https://app.grammarly.com/. It has a plugin for the browser.

Reply:
Thanks. It will help a lot.